### PR TITLE
479 import sizes of plants and squash

### PR DESCRIPTION
**Pull request by temmey**
## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [x] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [x] I fixed all affected decisions
- [ ] I added unit tests for my code
- [x] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [x] I mentioned [every code or binary](https://github.com/ElektraInitiative/PermaplanT/blob/master/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [x] I created left-over issues for things that are still to be done
- [x] Code is conforming to [our Architecture](/doc/architecture)
- [x] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [x] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+
+ALTER TABLE ONLY relations
+    ADD CONSTRAINT relations_pkey PRIMARY KEY (plant1, plant2);
+
+
+ALTER TABLE ONLY seeds
+    ADD CONSTRAINT seeds_pkey PRIMARY KEY (id);
+
+
+CREATE TRIGGER check_layer_type_before_insert_or_update BEFORE INSERT OR UPDATE ON plantings FOR EACH ROW EXECUTE FUNCTION check_layer_type();
```
Comment (markus2330) [inline backend/migrations/2023-07-15-152628_squashed_setup/up.sql:438]: This is a bit out-of-place here. The rest of the file is grouped differently.

File: doc/CHANGELOG.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ Syntax: `- short text describing the change _(Your Name)_`
 - _()_
 - _()_
 - _()_
-- _()_
+- squashed migrations and commented out properties for now to improve build time. Also set all enums to lowercases _(temmey)_
```
Comment (markus2330) [inline doc/CHANGELOG.md:45]: better make it 3 lines as it is unrelated

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql
Comment (markus2330) [inline backend/migrations/2023-07-15-152628_squashed_setup/up.sql:1]: Did you already format it according #233? Please also add to release notes.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job, we love many parts of this PR! :rocket:

The "Could probably also be removed?" means: if it is not used in code yet, please comment  it out, too.

Please create follow up issue if you want to update ER diagram in different PR.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+    y integer NOT NULL,
+    width integer NOT NULL,
+    height integer NOT NULL,
+    rotation real NOT NULL,
+    scale_x real NOT NULL,
+    scale_y real NOT NULL,
+    add_date date,
+    remove_date date,
+    create_date date DEFAULT CURRENT_DATE NOT NULL,
+    delete_date date
```
[the remark this answers is not in the document]
Comment (Bushuo) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:354]: `add_date` and `remove_date` are used in the timeline feature.

`create_date` and `delete_date` are currently not used.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+    flower_type flower_type,
+    created_at timestamp without time zone DEFAULT now() NOT NULL,
+    updated_at timestamp without time zone DEFAULT now() NOT NULL,
+    has_drought_tolerance boolean,
+    tolerates_wind boolean,
+    plant_references text[],
+    is_tree boolean,
+    nutrition_demand nutrition_demand,
+    preferable_permaculture_zone smallint,
+    article_last_modified_at timestamp without time zone,
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:253]: for preferable_permaculture_zone we have a constraint `CONSTRAINT plant_detail_preferable_permaculture_zone_check CHECK (((preferable_permaculture_zone IS NULL) OR ((preferable_permaculture_zone >= '-1'::integer) AND (preferable_permaculture_zone <= 6))))`, can we disable both?

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 5]
```diff
@@ -0,0 +1,461 @@
...
+    growth_rate growth_rate[],
+    height plant_height,
+    fertility fertility[],
+    --flower_colour text,
+    flower_type flower_type,
+    created_at timestamp without time zone DEFAULT now() NOT NULL,
+    updated_at timestamp without time zone DEFAULT now() NOT NULL,
+    has_drought_tolerance boolean,
+    tolerates_wind boolean,
+    plant_references text[],
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:249]: update_at can be disabled, but it is filled with data 100% because the insert_plants scraper updates these values when it changes values, i would keep it.
in `src/model/entity.rs`:
has_drought_tolerance is declared as used in hydrology layer
tolerates_wind is declared as used in wind layer

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 6]
```diff
@@ -0,0 +1,461 @@
...
+    id integer NOT NULL,
+    unique_name text NOT NULL,
+    common_name_en text[],
+    common_name_de text[],
+    family text,
+    genus text,
+    edible_uses_en text,
+    --medicinal_uses text,
+    material_uses_and_functions text,
+    botanic text,
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:225]: edible_uses_en text is used for search functionality

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 7]
```diff
@@ -0,0 +1,461 @@
...
+    --native_environment text,
+    ecosystem_niche text,
+    deciduous_or_evergreen deciduous_or_evergreen,
+    herbaceous_or_woody herbaceous_or_woody,
+    life_cycle life_cycle[],
+    growth_rate growth_rate[],
+    height plant_height,
+    fertility fertility[],
+    --flower_colour text,
+    flower_type flower_type,
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: height will be used by me in the next few days, the import scraper for the mergedata.csv is already done. 
fertility & flower_type is used in `backend/src/schema.patch` and cant be removed. Is it important @ThorbenStaufer?

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 1 above)
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql]: i moved it to the bottom of the file for now.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 8]
```diff
@@ -0,0 +1,461 @@
...
+CREATE TABLE plants (
+    id integer NOT NULL,
+    unique_name text NOT NULL,
+    common_name_en text[],
+    common_name_de text[],
+    family text,
+    genus text,
+    edible_uses_en text,
+    --medicinal_uses text,
+    material_uses_and_functions text,
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:224]: edible_uses_en is used for search

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:249]: i will comment out tolerates_wind for now.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql [hunk 9]
```diff
@@ -0,0 +1,461 @@
...
+);
+
+CREATE TYPE plant_spread AS ENUM (
+    'narrow',
+    'medium',
+    'wide',
+    'na'
+);
+
+CREATE TYPE privacy_options AS ENUM (
```
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:91]: or calling it `privacy_option`?

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:1]: no, i would like to use the linter directly for that. Can we merge #233 or should i cherry pick the commit into this branch to test it? Or should i make a new issue out of it?

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 7 above)
[the remark this answers is not in the document]
Comment (ThorbenStaufer) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: > Is it important

Yes, the patch file is very important for our PostGis integration to work.
But it can simply be adapted or re-generated in your case, since fertility and flower_type are not crucial parts of it.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql
Comment (markus2330) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:1]: Better you cherry-pick from #233, then we don't have a gap where changing existing SQL files might lead to unwanted reformatting. But we can also do the formatting as follow-up after this PR.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (markus2330) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql]: ok, lets keep it. `tolarates_wind` is needed for wind layer.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (markus2330) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql]: The comment is on `botanic`.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 4 above)
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql]: I assumed you wanted me to disabled preferable_permaculture_zone too but I will keep it for now.

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:1]: done with [a0e6333](https://github.com/ElektraInitiative/PermaplanT/pull/644/commits/a0e633366772695d3dafc8c0bf14ab71f9ccc3d7)

File: backend/migrations/2023-07-15-152628_squashed_setup/up.sql (hunk 7 above)
[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: since it also generates issue for the down.sql script, lets discuss it in another issue if and how we gonna re-generate it. The database still can be reset with `diesel database reset`

[the remark this answers is not in the document]
Comment (ThorbenStaufer) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: `database reset` doesn't do anything for the issues that arise if we don't have a functioning `schema.patch`.

[the remark this answers is not in the document]
Comment (temmey) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: sorry i missed an important part. Right now `diesel migration redo` throws `error applying hunk #1` but clears `schema.rs`. i need to change `schema.patch` im just not sure how yet

[the remark this answers is not in the document]
Comment (ThorbenStaufer) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: The patch file itself is just a git diff file, that is read by diesel to make custom adaptations to the schema file.
There are two ways you could fix it:
1. manually update the patch file, so it can correctly detect the location in the schema file, that needs to be updated.
2. delete the old patch file, run diesel to get a filled schema file, manually make the changes in the schema file and then run `git diff -U6 > src/schema.patch` to let git generate a new diff file from your manual changes.

[the remark this answers is not in the document]
Comment (ThorbenStaufer) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: Addendum to solution 1:
Manually updating the patch file is a bit tricky at first. Not only do you need to make sure that the code snippet inside the patch file perfectly matches the schema file how it would be generated by diesel without the patch file, but you also need to update the line count in the diff header. Without that diesel will not recognize it as a valid git diff.

[the remark this answers is not in the document]
Comment (ThorbenStaufer) [reply backend/migrations/2023-07-15-152628_squashed_setup/up.sql:244]: Addendum to solution 2:
For git to detect manual changes to the schema file, you need to temporarily remove it from git ignore and commit it to your local repository. It is important, that you first commit it and only after that make manual changes, since the git diff file would include the entire schema otherwise. Just don't forget to include it in gitignore again later and commit it's deletion.

**Review by markus2330 — APPROVED** [review]
Maybe we merge it earlier as thought.

**Comment by markus2330** [comment]
@temmey there are some frontend errors here. Can you fix it?

**Comment by markus2330** [comment]
But we stay with the plan discussed today, first please the UI fixes (names of plants)

**Comment by temmey** [comment]
> @temmey there are some frontend errors here. Can you fix it?
i managed to fix a few

**Comment by temmey** [comment]
Hi @markus2330, can we merge this now? This is a big update and other approved changes are adding onto this, making it more complicated.

**Comment by markus2330** [comment]
Thx, great work.

@temmey there is a "please merge" label to indicate this.

**Comment by 4ydan** [comment]
Amazing, pipeline is so fast now.

**Comment by markus2330** [comment]
Thanks to you both :heart:
